### ISSUE TITLE
Can't see unicode characters

### ISSUE THREAD

**Issue by cagdassalur** [FelixKratz/SketchyBar#154]
Here's the output from my scripts and on the bottom my tmux status bar showing correct characters:
![image](https://user-images.githubusercontent.com/3230608/149543117-d282726b-8dd0-4227-b4b6-cb01dc0edca2.png)
However on SketchyBar i see the '_' character instead for the same script:
![image](https://user-images.githubusercontent.com/3230608/149543320-3f2772b4-e5d4-4025-8499-b1902a9c4a65.png)

Is there a setting to enable utf-8? Right now i'm using the default fonts for icons and labels.

Also thank you for making this awesome bar!

**Comment by FelixKratz** [comment]
I have tested some unicode characters and they appear to be working for me, can you try setting some item like this:
```bash
sketchybar --set <name> label=🔵🔴🔵🔵
```
does it show up in the bar now?

**Comment by cagdassalur** [comment]
The colored dots show up without an issue. I wonder what is wrong with my characters.. They show up fine in terminal..
Even in github: ✓ ↓8.0KiB/s • ↑1.8KiB/s

Oh i just noticed if i copy again from above text and set label manually it works.. My original characters has a problem.. I guess i can replace them with the chars i copied from this comment. weird..
Thanks for the amaszingly fast answer!

**Comment by cagdassalur** [comment]
Okay after some investigation i have some info: 
When i run this directly from command line it works:  
`sketchybar --set network label=$(~/.tmux/plugins/tmux-online-status/scripts/online_status_icon.sh)`

But when i have this sh file and provided through --set name script i get the "_" character.. Is there a difference on how this script is being run?

sh file:
```
#!/usr/bin/env bash
sketchybar --set network label=$(~/.tmux/plugins/tmux-online-status/scripts/online_status_icon.sh)

```
Item setting from sketchybarrc:
```
	# NETWORK
	sketchybar -m --add item network right \
		--set network label.color=0xffeceff4 \
		--set network update_freq=5 \
		--set network script="~/.config/sketchybar/plugins/tmux/network.sh" \

```

**Comment by cagdassalur** [comment]
Simplified the example:
test.sh
```
#!/usr/bin/env zsh
echo "✓🔵🔴🔵🔵"
```

network.sh (which is set as script of bar item)
```
#!/usr/bin/env zsh
NAME="network"
sketchybar --set network label=$(/Users/cagdassalur/.config/sketchybar/plugins/tmux/test.sh)
```

When i run network.sh from my shell i can see the characters correctly, but when it updates through sketchybar after couple seconds then it shows '_'

**Comment by es183923** [comment]
I've had this problem before with reverse colorization. Does the output have colorization or reverse coloring?

**Comment by es183923** [comment]
Does this occur if you manually run network.sh?

**Comment by cagdassalur** [comment]
> I've had this problem before with reverse colorization. Does the output have colorization or reverse coloring?

I'm not sure what you mean by reverse colorization, i don't see the characters at all (see the screenshots).. It's showing as '_'

And no if i run network.sh from my shell i can see everything
manual run of network.sh:
<img width="120" alt="image" src="https://user-images.githubusercontent.com/3230608/149569087-df5564b5-ae2f-461f-9867-68b1a4ba02a8.png">

After upated automatically through update_freq:
<img width="53" alt="image" src="https://user-images.githubusercontent.com/3230608/149569165-8fcb8a1b-cf41-49bd-8b89-5aa25d6ac303.png">

**Comment by es183923** [comment]
in that case, I'm not sure what's happening.

**Comment by FelixKratz** [comment]
> Simplified the example: test.sh
> 
> ```
> #!/usr/bin/env zsh
> echo "✓🔵🔴🔵🔵"
> ```
> 
> network.sh (which is set as script of bar item)
> 
> ```
> #!/usr/bin/env zsh
> sketchybar --set network label=$(/Users/cagdassalur/.config/sketchybar/plugins/tmux/test.sh)
> ```
> 
> When i run network.sh from my shell i can see the characters correctly, but when it updates through sketchybar after couple seconds then it shows '_'

This is very odd. I have tried your exact example and it seems to be working just fine. The only difference executing from shell vs in sketchybar is probably that your shell is zsh and sketchybar executes all scripts with sh. You could try running the script in shell via: sh tests.sh and see if that also produces the artifact.

Maybe try using another text editor to save the script file is all I can think of currently, but I will investigate further later today.

**Comment by cagdassalur** [comment]
running with sh also produces the same result, works from my shell but not sketchybar.. Tried with 
different editor as well. My guess would be that when c captures the script output when running with exec_vp somehow messes up the characters, but then it's working on your side, so maybe not that.. 
Did you run network.sh through item script property? without running it manually? Do you have the hack_nerd font? 

Not sure if it'll help but i'm attaching my sketchybarrc as well (added .txt extension so github accepts it)
[sketchybarrc.txt](https://github.com/FelixKratz/SketchyBar/files/7874681/sketchybarrc.txt)

**Comment by FelixKratz** [comment]
This is the minimal configuration I am using to test this, try placing this in your `.config` folder.
[sketchybar.zip](https://github.com/FelixKratz/SketchyBar/files/7874767/sketchybar.zip)

It produces the following output with sketchybar v2.4.0:
 
<img width="1512" alt="Screen Shot 2022-01-15 at 11 35 23" src="https://user-images.githubusercontent.com/22680421/149618690-e6857af8-a5bb-4bee-a7ef-4e41f3e19220.png">

**Comment by cagdassalur** [comment]
This is really weird.. With your test script it works, so i'm guessing something is wrong with the script i'm using, but i still don't understand how it works when i run the command through shell i see them..

**Comment by cagdassalur** [comment]
I think this can be closed, i have to fix this from my side somehow.. I'll update this issue if i'm ever able to fix it for future reference for others.

**Comment by FelixKratz** [comment]
One thing I was thinking is what does `locale` print in terminal vs sketchybar? For me it is:
```bash
LANG=""
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```
for both sketchybar and the terminal. The UTF-8 bit is important I think.

**Comment by cagdassalur** [comment]
This might be it, it was LC ones were 'C' so i set them properly for my terminal, changed nothing but here's the interesting part: When i put echo \`locale\` in test.sh in sketchybar.out.log i see this:
```
Invalid propery: LANG
Invalid propery: LC_COLLATE
Invalid propery: LC_CTYPE
Invalid propery: LC_MESSAGES
Invalid propery: LC_MONETARY
Invalid propery: LC_NUMERIC
Invalid propery: LC_TIME
Invalid propery: LC_ALL
```

**Comment by FelixKratz** [comment]
If you stop the brew service: `brew services stop sketchybar` you can run sketchybar directly from your terminal via: `sketchybar` and you will be able to see all script outputs, so just add a script that does:
```bash
echo "$(locale)"
```
and you will see the output printed in the terminal.

**Comment by cagdassalur** [comment]
Very strange.. I ran the sketchybar directly from my terminal to see locale output and discovered that it actually shows the characters when i do that. It doesn't show them when i do `brew services stop sketchybar`

**Comment by FelixKratz** [comment]
It seems that changing the system language to English (US) fixes the problem, it is very odd that macOS does this for some language combinations https://github.com/Homebrew/homebrew-core/issues/41139. Maybe it is best to set the `LANG` variable in the homebrew formula https://github.com/FelixKratz/homebrew-formulae/blob/master/sketchybar.rb explicitly, however I am not sure if that produces other artifacts that I am not aware of.

**Comment by cagdassalur** [comment]
Thanks! `defaults read -g AppleLocale` returns "en_TR" for me, you might be right about some language combinations. I guess fix is to set it from formula for now.. 
Tip about the locale was really helped a lot, thanks for debugging this with me.

**Comment by FelixKratz** [comment]
I have included this in the new release f0b6154e93cc87865e462caf7b2d1d574dc61e7a
